Text/CustomEntitySearch/CustomEntitySearch.cs

```diff
@@ -95,5 +128,310 @@ public static class CustomEntitySearch
             return new OkObjectResult(response);
         }
 
+        public static void AddValues(string checkMatch, string text, List<Entity> entities, 
+            HashSet<string> entitiesFound, int leniency, bool caseSensitive)
+        {
+            if (leniency == 0)
+            {
+                string escapedWord = Regex.Escape(checkMatch);
+                string pattern = (caseSensitive) ? @"(?x:" + escapedWord + @")" : @"(?ix:" + escapedWord + @")";
+                if (!escapedWord.First().IsDelineating() && !substringMatch)
+                    pattern = @"\b" + pattern;
+                if (!escapedWord.Last().IsDelineating() && !substringMatch)
+                    pattern += @"\b";
+                MatchCollection entityMatch = Regex.Matches(text, pattern, RegexOptions.Compiled, TimeSpan.FromSeconds(MaxRegexEvalTime));
+                if (entityMatch.Count != 0)
+                {
+                    foreach (Match match in entityMatch)
+                    {
+                        entities.Add(
+                            new Entity
+                            {
+                                Category = "customEntity",
+                                Value = match.Value,
+                                Offset = match.Index,
+                                Confidence = 1
+                            });
+                    }
+                    entitiesFound.Add(checkMatch);
+                }
+            }
+            else
+            {
+                // Begin searching!
+                int whitespaceOffset = 0;
+                double bestMismatchPreWhitespace = -1;
+                int prevWhiteSpaceIndex = 0;
+                int currTextCharIndex = 0;
+                int currWordCharIndex = 0;
+                StringWriter wordFound = new StringWriter();
+                double currMismatch = 0;
+                string wordCharArray = (caseSensitive) ? CreateWordArray(checkMatch) : CreateWordArray(checkMatch.ToLower(CultureInfo.CurrentCulture));
+                string textCharArray = (caseSensitive) ? text : text.ToLower(CultureInfo.CurrentCulture);
+                if (leniency >= wordCharArray.Length)
+                    leniency = wordCharArray.Length - 2;
+
+                if (leniency >= 0)
+                {
+                    while (currTextCharIndex < textCharArray.Length)
+                    {
+                        // First find the delineating character for prefix addition later on
+                        if ((currWordCharIndex == 0 || currMismatch > 0) && textCharArray[currTextCharIndex].IsDelineating())
```

Comment:
> textCharArray[currTextCharIndex].IsDelineating() [](start = 75, length = 49)

nit: you can combine this conditional and the one below it into one dealing with delineating characters